File: java/lib/src/main/java/com/svix/models/BackgroundTaskStatus.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.Utils.ToQueryParam;

public enum BackgroundTaskStatus implements ToQueryParam {
    RUNNING("running"),
    FINISHED("finished"),
    FAILED("failed");
    private final String value;

    BackgroundTaskStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: php/src/Models/BackgroundTaskStatus.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

enum BackgroundTaskStatus: string implements \JsonSerializable
{
    case RUNNING = 'running';
    case FINISHED = 'finished';
    case FAILED = 'failed';

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return self::from($data);
    }

    public function jsonSerialize(): mixed
    {
        return $this->value;
    }
}
